src/QsCompiler/CommandLineTool/CompilationTracker.cs

```diff
@@ -0,0 +1,281 @@
+                foreach (KeyValuePair<string, CompilationTask> entry in CompilationTasks)
+                {
+                    if (parentNode.Task.Name.Equals(entry.Value.ParentName))
+                    {
+                        CompilationTaskNode childNode = new CompilationTaskNode(entry.Value);
+                        parentNode.Children.Add(childNode.Task.Name, childNode);
+                        toFindChildrenNodes.Enqueue(childNode);
+                    }
+                }
+            }
+
+            return compilationTasksForest;
+        }
+
+        /// <summary>
+        /// Handles a compilation task start event.
+        /// </summary>
+        private static void CompilationEventStartHandler(CompilationLoader.CompilationTaskEventArgs eventArgs)
+        {
+            string key = CompilationTask.GenerateKey(eventArgs.ParentTaskName, eventArgs.TaskName);
+            if (CompilationTasks.ContainsKey(key))
+            {
+                Warnings.Add(new Warning(WarningType.ProcessAlreadyExists, key));
+                return;
+            }
+
+            CompilationTasks.Add(key, new CompilationTask(eventArgs.ParentTaskName, eventArgs.TaskName));
+        }
+
+        /// <summary>
+        /// Handles a compilation task end event.
+        /// </summary>
+        private static void CompilationEventEndHandler(CompilationLoader.CompilationTaskEventArgs eventArgs)
+        {
+            string key = CompilationTask.GenerateKey(eventArgs.ParentTaskName, eventArgs.TaskName);
+            if (!CompilationTasks.TryGetValue(key, out CompilationTask process))
+            {
+                Warnings.Add(new Warning(WarningType.ProcessDoesNotExist, key));
+                return;
+            }
+
+            if (!process.IsInProgress())
+            {
+                Warnings.Add(new Warning(WarningType.ProcessAlreadyEnded, key));
+                return;
+            }
+
+            process.End();
+        }
+
+        // Public methods.
+
+        /// <summary>
+        /// Handles a compilation task event.
+        /// </summary>
+        public static void OnCompilationTaskEvent(object sender, CompilationLoader.CompilationTaskEventArgs args)
+        {
+            CompilationEventTypeHandlers[args.Type](args);
+        }
+
+        /// <summary>
+        /// Publishes the results to text files in the specified folder.
+        /// </summary>
+        public static void PublishResults(string outputFolder)
+        {
+            IList<CompilationTaskNode> compilationProcessesForest = BuildCompilationTasksHierarchy();
+            string outputPath = Path.GetFullPath(outputFolder);
+            DirectoryInfo outputDirectoryInfo = new DirectoryInfo(outputPath);
+            if (!outputDirectoryInfo.Exists)
+            {
+                outputDirectoryInfo.Create();
+            }
+
+            using (StreamWriter file = File.CreateText(Path.Combine(outputPath, "compilationPerf.json")))
```

Comment:
Consider using the 'var' keyword for 'using' block variables.
```suggestion
            using (var file = File.CreateText(Path.Combine(outputPath, "compilationPerf.json")))
```